File: plugins/itp-hooks/tests/fixtures/js/good_code.js
// Test fixture: GOOD JS code (should pass)
async function goodFunction() {
    try {
        await riskyOperation();
    } catch (e) {
        console.error('Operation failed:', e);
        throw e;
    }
}

function goodPromise() {
    fetchData()
        .then(data => console.log(data))
        .catch(err => console.error('Fetch failed:', err));
}

File: plugins/itp-hooks/tests/fixtures/python/good_code.py
# Test fixture: GOOD Python code (should pass)
import subprocess
import logging

logger = logging.getLogger(__name__)

def good_function():
    try:
        risky_operation()
    except ValueError as e:
        logger.error("Operation failed: %s", e)
        raise

def good_subprocess():
    subprocess.run(["ls", "-la"], check=True)
